virtualmachine/vsphere/util.go

```diff
@@ -603,7 +669,22 @@ var createTemplateName = func(t string, ds string) string {
 var uploadTemplate = func(vm *VM, dcMo *mo.Datacenter, selectedDatastore string) error {
 	template := createTemplateName(vm.Template, selectedDatastore)
 	vm.datastore = selectedDatastore
+	basePath, err := ioutil.TempDir("", "")
+	if err != nil {
+		return err
+	}
+	defer func() {
+                if err := os.RemoveAll(basePath); err != nil {
+                        fmt.Errorf("can't remove temp directory, error: %s", err.Error())
+                }
+        }()
 	// Read the ovf file
```

Comment:
I think if both OvaPathUrl and OvfPath is specified, Ovapath will override the Ovfpath. Is that correct?
I think only if OvfPath is not specified or .Ovf is missing at specified OvfPath, we should download it. Please confirm, the behavior.